### PR TITLE
Integrated the changes from dome9

### PR DESCRIPTION
**Pull request by bcoddens**
What
Our codebuild environment needs double sources as input to build
Why
We use code that is based on gitlab , gitlab is not supported by codebuild so we need to upload zip files to support our codebuilds. We want to keep the config and source in a different upload.
These changes support using multiple s3 buckets as source
References
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codebuild_project#secondary_sources
Proceed on pull request: #73

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -367,6 +367,23 @@ resource "aws_codebuild_project" "default" {
     }
   }
 
+  dynamic "secondary_sources" {
+    iterator = secondary_source
+    for_each = length(var.secondary_sources) > 0 ? var.secondary_sources : []
```
Comment (jamengual) [inline main.tf:372]: you can delete the condition and it will null if secondary sources is empty

**Review by jamengual — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (bcoddens) [reply main.tf]: Changed pushed, is this what you mean ?

File: main.tf [hunk 2]
```diff
@@ -367,6 +367,22 @@ resource "aws_codebuild_project" "default" {
     }
   }
 
+  dynamic "secondary_sources" {
+    for_each = var.secondary_sources ? [""] : []
```
Comment (jamengual) [inline main.tf:371]: you can just do `for_each = var.secondary_sources`

**Review by jamengual — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (bcoddens) [reply main.tf]: Done

File: main.tf (hunk 2 above)
Comment (bcoddens) [reply main.tf]: Done

**Review by jamengual — DISMISSED** [review]

**Review by jamengual — APPROVED** [review]

**Pull request merged**